.github/workflows/e2e-tests.yaml

```diff
@@ -0,0 +1,42 @@
+name: e2e
+on:
+  pull_request:
```

Comment:
shouldn't we run this before the PR gets merged? 
It may be worth taking advantage of the `build` workflow to run the same in parallel